### PR TITLE
Cleanup controller ent

### PR DESCRIPTION
**Pull request by lkysow**
Changes proposed in this PR:
- Test enterprise support for cleanup controller
- companion to https://github.com/hashicorp/consul-k8s/pull/435

How I've tested this PR:
- acceptance tests

How I expect reviewers to test this PR:
- code review

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -155,7 +155,7 @@ jobs:
                     -kubecontext="kind-dc1" \
                     -secondary-kubecontext="kind-dc2" \
                     -debug-directory="$TEST_RESULTS/debug" \
-                    -consul-k8s-image=docker.mirror.hashicorp.services/hashicorpdev/consul-k8s:latest
+                    -consul-k8s-image=ghcr.io/lkysow/consul-k8s-dev:feb11
```
Comment (lkysow) [inline .circleci/config.yml:158]: todo: swap back

File: test/acceptance/tests/connect/connect_inject_test.go [hunk 2]
```diff
@@ -141,20 +141,8 @@ func TestConnectInject_CleanupController(t *testing.T) {
 			ns := ctx.KubectlOptions(t).Namespace
 			pods, err := ctx.KubernetesClient(t).CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{LabelSelector: "app=static-client"})
 			require.NoError(t, err)
```
Comment (lkysow) [inline test/acceptance/tests/connect/connect_inject_test.go:143]: After #818 we don't need to check for previous pods' installations.

File: test/acceptance/tests/connect/connect_inject_namespaces_test.go [hunk 3]
```diff
@@ -178,3 +182,120 @@ func TestConnectInjectNamespaces(t *testing.T) {
...
+				Namespace:   staticClientNamespace,
+			}
+			k8s.DeployKustomize(t, staticClientOpts, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-namespaces")
+
+			logger.Log(t, "waiting for static-client to be registered with Consul")
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+			expectedConsulNS := staticClientNamespace
+			if !c.mirrorK8S {
+				expectedConsulNS = c.destinationNamespace
+			}
```
Comment (thisisnotashwin) [inline test/acceptance/tests/connect/connect_inject_namespaces_test.go:263]: Isn't `c.destinationNamespace` always equal to `staticClientNamespace`? Wouldn't the `expectedConsulNS` always be `staticClientNamespace`?

**Review by thisisnotashwin — COMMENTED** [review]
This looks good. I might be misreading some of the code but it appears that the consul destination namespace is always `ns-2` regardless of mirroring or not. This could be reviewer error but what am i missing?

File: test/acceptance/tests/connect/connect_inject_namespaces_test.go (hunk 3 above)
Comment (lkysow) [reply test/acceptance/tests/connect/connect_inject_namespaces_test.go:264]: Oops, yeah! Fixing...

**Review by thisisnotashwin — APPROVED** [review]

**Review by kschoche — APPROVED** [review]

**Pull request merged**